src/python/zquantum/core/measurement.py

```diff
@@ -500,7 +504,30 @@ def get_expectation_values(
                     value = -float(count) / num_measurements
                 expectation += np.real(coefficient) * value
             expectation_values.append(np.real(expectation))
-        return ExpectationValues(np.array(expectation_values))
+
+        correlations = np.zeros((len(ising_operator.terms),) * 2)
+        for i, first_term in enumerate(ising_operator.terms):
+            for j, second_term in enumerate(ising_operator.terms):
```

Comment:
I would like to suggest a few minor optimizations here:
- We could exclude diagonal correlations since those are squared Ising operator coefficients.
- There is no need to calculate the correlations of the constant term with the others; they can be efficiently extracted from the expectation values of the terms and corresponding coefficients.
- \<Ai Aj\> = \<Aj Ai\> so we could loop over (i,j) pairs such that e.g. i < j.

Reply:
Updated this in the `precisions` branch